client/post-editor/media-modal/edit/edit-canvas.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+var React = require( 'react' ),
+	ReactDOM = require( 'react-dom' );
+
+/**
+ * Internal dependencies
+ */
+var noop = require( 'lodash/noop' );
+
+module.exports = React.createClass( {
+	displayName: 'MediaModalDetailEditCanvas',
+
+	propTypes: {
+		src: React.PropTypes.string,
+		rotate: React.PropTypes.number,
+		scaleX: React.PropTypes.number,
+		scaleY: React.PropTypes.number
+	},
+
+	getDefaultProps: function () {
+		return {
+			rotate: 0,
+			scaleX: 1,
+			scaleY: 1
+		}
+	},
+
+	getInitialState: function () {
+		return {
+			canvasWidth: this.props.item ? this.props.item.width : 0,
+			canvasHeight: this.props.item ? this.props.item.height : 0
+		};
+	},
+
+	componentWillReceiveProps: function ( newProps ) {
+		if ( this.props.src !== newProps.src ) {
+			this.initImage( newProps.src );
+		}
+	},
+
+	componentDidMount: function () {
+		if ( ! this.props.src ) {
+			return;
+		}
+
+		this.initImage( this.props.src );
+	},
+
+	initImage: function ( src ) {
+		this.image = new Image();
+		this.image.src = src;
+		this.image.onload = this.onLoadComplete;
+		this.image.onerror = this.onLoadComplete;
+	},
+
+	onLoadComplete: function ( event ) {
+		if ( event.type !== 'load' ) {
+			return;
+		}
+
+		this.drawImage();
+	},
+
+	componentDidUpdate: function () {
+		this.drawImage();
+	},
+
+	toBlob: function ( callback ) {
+		var canvas = ReactDOM.findDOMNode( this.refs.canvas );
+
+		canvas.toBlob( callback, "image/jpeg", 0.95 ); // JPEG at 95% quality
```

Comment:
Hi @mtias, this is just for initial presentation at the moment, but the ideal plan is to make it save in the same format as the uploaded image